engines/scumm/players/player_nes.cpp

```diff
@@ -127,8 +127,7 @@ static const int8 Duties[4][8] = {
 };
 
 void Square::Reset() {
-	memset(this, 0, sizeof(*this));
-	Cycles = 1;
+	*this = Square();
 	EnvCtr = 1;
 	BendCtr = 1;
```

Comment:
These two should also be initialized with appropriate default values, like `Cycles` above